lib/parseSty.js

```diff
@@ -6,33 +6,9 @@ var starts = ["\\newcommand",
               "\\defSpecFun"];
 
 function findBracketSections(string) {
-    var sections = [[""]];
-    var char;
-    var level = 0;
-    for (var i = 0; i < string.length; i++) {
-        char = string.charAt(i);
-        if (char === "}" || char === "]") {
-            level--;
-            if (level === 0) {
-                sections.push([""]);
-            }
-        }
-        else if (char === "{" || char === "[") {
-            if (level === 0) {
-                sections[sections.length - 1].push(char);
-            }
-            else {
-                sections[sections.length - 1][0] += char;
-            }
-            level++;
-            continue;
-        }
-        if (level !== 0) {
-            sections[sections.length - 1][0] += char;
-        }
-    }
-    sections = sections.slice(0, sections.length - 1);
-    return sections;
+    return string.match(/\{[^\{\}]*?(([^\}]*)\}+)*?[^\{\}]*?\}|\[[^\[\]]*?(([^\]]*)\]+)*?[^\[\]]*?\]/g).map(function(s) {
```

Comment:
In Germany one calls that "to reinvent the wheel".
How about using a well tested library for that like
https://www.npmjs.com/package/balanced-match
?